devel/create_distro_tarball.sh

```diff
@@ -0,0 +1,187 @@
+  echo "examples of git_tag  would be 'master' or '1.11.33.4'"
+  exit 1
+}
+
+# This outputs a little wrapper around gyp that calls it with appropriate -D
+# flag
+function config {
+  cat <<SCRIPT_END
+#!/bin/sh
+#
+# This script uses gyp to generate Makefiles for mod_pagespeed built against
+# the following system libraries:
+#   apr, aprutil, apache httpd headers, icu, libjpeg_turbo, libpng, zlib.
+#
+# Besides the -D use_system_libs=1 below, you may need to set (via -D var=value)
+# paths for some of these libraries via these variables:
+#   system_include_path_httpd, system_include_path_apr,
+#   system_include_path_aprutil.
+#
+# for example, you might run
+# ./generate.sh -Dsystem_include_path_apr=/usr/include/apr-1 \\
+#               -Dsystem_include_path_httpd=/usr/include/httpd
+# to specify APR and Apache include directories.
+#
+# Also, BUILDTYPE=Release can be passed to make (the default is Debug).
+echo "Generating src/Makefile"
+src/build/gyp_chromium -D use_system_libs=1 \$*
+SCRIPT_END
+}
+
+if [ $# -lt 2 ]; then
+  usage
+  exit
+fi
+
+CHECKOUT_DIR=
+if [ "$1" = "--checkout_dir" ]; then
+  if [ -z $2 ]; then
+    usage
+  fi
+  CHECKOUT_DIR=$2
+  shift 2
+else
+  CHECKOUT_DIR=`mktemp -d`
+  echo $CHECKOUT_DIR
+fi
+
+TARBALL=$1
+if [ -z $TARBALL ]; then
+  usage
+fi
+touch $TARBALL
+TARBALL=`realpath $TARBALL`
+
+
+if [ "$2" == --git_tag ]; then
+  if [ -z $3 ]; then
+    usage
+  else
+    GIT_TAG=$3
+    if [ -z $CHECKOUT_DIR ]; then
+      CHECKOUT_DIR=`mktemp -d`
+    fi
+    SRC_DIR=$CHECKOUT_DIR
+    cd $CHECKOUT_DIR
+    git clone https://github.com/pagespeed/mod_pagespeed.git src\
+      --branch "$GIT_TAG" --recursive
+    cd src
+  fi
+else
+  SRC_DIR=$2
+fi
+
+cd $SRC_DIR
+
+# Pick up our version info, and wrap src inside a modpagespeed-version dir,
+# so unpackers don't get a src/ surprise
+source src/net/instaweb/public/VERSION
+VER_STRING=$MAJOR.$MINOR.$BUILD.$PATCH
+DIR=modpagespeed-$VER_STRING
+rm -rf $DIR
+mkdir $DIR
+echo ln -sf $SRC_DIR/src $DIR/src/
+ln -sf $SRC_DIR/src $DIR/src
+
+# Also create a little helper script that shows how to run gyp
+config > $DIR/generate.sh
+chmod +x $DIR/generate.sh
+
+# Normally, the build system runs build/lastchange.sh to figure out what
+# to put into the last portion of the version number. We are, however, going to
+# be getting rid of the .svn dirs, so that will not work (nor would it without
+# network access). Luckily, we can provide the number via LASTCHANGE.in,
+# so we just compute it now, and save it there.
+cd src/
+./build/lastchange.sh $PWD > LASTCHANGE.in
+cd ..
+
+# It's tarball time!
+# TODO(morlovich): Fix giflib.gyp to support proper external depends.
+# Note that this is highly-version specific, and may require tweaks for
+# every release as its dependencies change. It's probably a good idea to
+# branch this script with every release.
+tar cj --dereference --exclude='.git' --exclude='.svn' --exclude='.hg' -f $TARBALL \
```

Comment:
`--exclude='.svn' --exclude='.hg'` shouldn't be needed anymore now that all our dependencies are git.